x/concentrated-liquidity/keeper_test.go

```diff
@@ -470,32 +472,106 @@ func (s *KeeperTestSuite) runMultiplePositionRanges(ranges [][]int64, rangeTestP
 // creates a pososition from the first test account with default coins and no slippage protection
 // for a given pool id and ticks.
 // a convinience method for testing.
-func (s *KeeperTestSuite) createDefaultPosition(poolId uint64, lowerTick, upperTick int64) uint64 {
+func (s *KeeperTestSuite) createDefaultPosition(poolId uint64, lowerTick, upperTick int64) cl.CreatePositionData {
 	positionData, err := s.App.ConcentratedLiquidityKeeper.CreatePosition(s.Ctx, poolId, s.TestAccs[0], DefaultCoins, sdk.ZeroInt(), sdk.ZeroInt(), lowerTick, upperTick)
 	s.Require().NoError(err)
-	return positionData.ID
+	return positionData
 }
 
 // sets up positions for testing the migration of the min spot price from 10^-12 to 10^-30.
 // Specifically, creates positions:
 // - original full range
 // - new extended full range
 // - position between the new min spot price and the old min spot price
-func (s *KeeperTestSuite) setupPositionsForMinSpotPriceMigration() uint64 {
-	pool := s.PrepareConcentratedPool()
-	poolId := pool.GetId()
+func (s *KeeperTestSuite) setupPositionsForMinSpotPriceMigration(spreadFactor osmomath.Dec) (poolId uint64, positions []cl.CreatePositionData) {
+	pool := s.PrepareCustomConcentratedPool(s.TestAccs[0], ETH, USDC, DefaultTickSpacing, spreadFactor)
+	poolId = pool.GetId()
 
 	// Fund test account with tokens necessary for all positions
 	s.FundAcc(s.TestAccs[0], DefaultCoins.Add(DefaultCoins...).Add(DefaultCoins...))
 
 	// Setup an original full range position
-	s.createDefaultPosition(poolId, types.MinInitializedTick, types.MaxTick)
+	position := s.createDefaultPosition(poolId, types.MinInitializedTick, types.MaxTick)
+	positions = append(positions, position)
 
 	// Setup a full range position on the new min spot price
-	s.createDefaultPosition(poolId, types.MinInitializedTickV2, types.MaxTick)
+	position = s.createDefaultPosition(poolId, types.MinInitializedTickV2, types.MaxTick)
+	positions = append(positions, position)
 
 	// Setup a position between the new min spot price and the old min spot price
-	s.createDefaultPosition(poolId, types.MinInitializedTickV2+1000, types.MinInitializedTick-1000)
+	position = s.createDefaultPosition(poolId, types.MinInitializedTickV2+1000, types.MinInitializedTick-1000)
+	positions = append(positions, position)
+
+	return poolId, positions
+}
+
+// swaps to the lowered minimum tick of 10^-30 and back to the original tick which is in the positive range.
+// There are 3 positions setup over the swapped range (see setupPositionsForMinSpotPriceMigration for details).
+// Additionally, spread factor and incentive rewards are configurable as parameters.
+// Returns pool id position data and actual tokens swapped in (in zero for one direction and back).
+func (s *KeeperTestSuite) swapToMinTickAndBack(spreadFactor osmomath.Dec, incentiveRewards sdk.Coins) (poolId uint64, positions []cl.CreatePositionData, actualAmountsSwappedIn sdk.Coins) {
```

Comment:
Note to reviewer: the majority of the setup is extracted from `TestSwap_MinSpotPriceMigration`